integration/helper/conflict.go

```diff
@@ -0,0 +1,30 @@
+package helper
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	kclient "github.com/acorn-io/acorn/pkg/k8sclient"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func DoUntilNoConflict(t *testing.T, ctx context.Context, c client.Client, obj client.Object, updater func()) error {
+	t.Helper()
+
+	for {
+		if err := c.Get(ctx, kclient.ObjectKey{Namespace: obj.GetNamespace(), Name: obj.GetName()}, obj); err != nil {
+			t.Fatal(err)
+		}
+
+		updater()
+
+		if err := c.Update(ctx, obj); !apierrors.IsConflict(err) {
+			return err
+		}
+
+		// Give things a chance to settle down.
+		time.Sleep(time.Second)
+	}
+}
```

Comment:
It looks like the only way this loop terminates is if an error occurs. I might be misunderstanding something, but should there be a check to break from the loop if the update succeeded?

Reply:
IsConflict returns false if `err == nil`.

Reply:
Oh that makes sense. For some reason my brain was looking at that return and thought that it was another error case and not a success case.